### DISCUSSION TITLE
macOS/FS-UAE config?

### DISCUSSION THREAD

**Discussion by DidierMalenfant** [BartmanAbyss/vscode-amiga-debug#234] [Q&A] [answered]
I'm trying to change the configuration used for FS-UAE on macOS. I mainly want to change the window properties, turn off floppy noise, etc...

In the README it says:
```
if you want to change some things (resolution, window size, etc.) just go into the Configurations tab, select default, and hit Save
```
I'm a VSCode noob and I'm not sure where this 'Configurations` tab is. Any pointers are appreciated!

**Comment by BartmanAbyss** [comment]
This section in the README only applies to WinUAE. I have no clue about FS-UAE. @grahambates ?

**Reply by DidierMalenfant** [reply]
Makes sense. I'm willing to help and do some research too if need be.

**Comment by DidierMalenfant** [comment]
It looks like on windows it loads a `default` config file but not on `macOS`:
```
try {
config = parseCfg(fs.readFileSync(defaultPath, 'utf-8'));
} catch(e) { /**/ }
```
Is this where user-defined options would be stored on Windows? If so I can attempt to mirror this on macOS.

**Comment by DidierMalenfant** [comment, marked as answer]
So it turns out the way to do this is to pass arguments via `emuargs` in the `launch.json` file located inside the project's `.vscode` folder. Like this:
```
        {
            "type": "amiga",
            "request": "launch",
            "preLaunchTask": "compile",
            "name": "Amiga 500",
            "config": "A500",
            "program": "${workspaceFolder}/${config:amiga.program}",
            "kickstart": "${config:amiga.rom-paths.A500}",
            "internalConsoleOptions": "openOnSessionStart",
            "emuargs": [
                "--floppy_drive_0_sounds=off",
                "--fast_memory=16384"
            ]
        },
```
